### DISCUSSION TITLE
IDE error after 1.13.x - Expected Connection to be a GraphQL Object type.

### DISCUSSION THREAD

**Discussion by alancwoo** [wp-graphql/wp-graphql#2723] [Q&A] [answered]
I'm not sure if this is a bug or a conflict somehow (though have tried to disable several other plugins with the same issue), but accessing the graphql IDE results in a console error: `app.js?ver=029d819e2c38c9f70120:38 Uncaught (in promise) Error: Expected Connection to be a GraphQL Object type.` caused by `https://domain.com/wp-content/plugins/wp-graphql/build/app.js?ver=029d819e2c38c9f70120`

I tried all rolling back through all versions from `1.13.8` all of which had the same error until I reached `1.12.3` which works again. Any newer version results in the above error. I am on `Wordpress 6.1.1`

I am also using `WPGraphQL WPML 1.1.0` and `WPGraphQL for Advanced Custom Fields 0.6.1` but disabling them has no effect. Reading through the [release notes for 1.13.0](https://github.com/wp-graphql/wp-graphql/releases/tag/v1.13.0) it mentions possible breaking changes but I can't seem to see how this could affect just loading the IDE.

In all cases, the graphql endpoint is still functioning, just the IDE is broken.

**Comment by justlevine** [comment]
Hey @alancwoo 

WPGraphQL v1.13.x began enforcing the Relay Connection spec, and that's likely what's causing your error. You mentioned you tried disabling (I'm hoping) _all_ your 3rd-party extensions, but what about custom code?

The big breaking change you need to look for is that GraphQL objects at the end of one of WPGraphQL's connection resolvers need to be able to implement the `Node` interface, and more specifically have an `id: ID` field that holds the global ID (and not something else e.g. an int with the WP id). If it's not coming from an outdated extension, then look at any custom type/field/connections you have registered.

The reason it's happening in your IDE is that the entire schema gets introspected, exposing the error that you'd otherwise only get if you were querying the specific (lazy-loaded) connection.

**Comment by justlevine** [comment, marked as answer]
Also, if you turn on GraphQL Debugging and inspect the network response from `?graphql`, you should get some more info to help pinpoint you to the culprit. Feel free to share a screenshot here if you need help deciphering it

**Reply by alancwoo** [reply]
Hi @justlevine thanks so much for the response, I'm sorry for the delay getting back to you here. I've tried with every single plugin disabled except WP GraphQL with the same issue.

The issue is likely with my theme which implements custom post types/taxonomies - as switching to a default theme allows the IDE to function again!

So I enabled debugging as you recommended and saw under `extensions > debug` a `DUPLICATE_TYPE` object and realized I had a custom post type and taxonomy set to the same graphql name. Adjusting this seems to have fixed the IDE.

Thanks for the help!
